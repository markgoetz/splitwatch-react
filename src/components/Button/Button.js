import React from 'react';
import PropTypes from 'prop-types';
import style from './Button.module.css';

export default function Button(props) {
    const baseClassName = style.button;
    const variantClassName = props.isPrimary ? style.primary : style.secondary;
    const className = [baseClassName, variantClassName].join(' ');
    return <button className={className} onClick={props.onClick}>{props.children}</button>;
}

Button.propTypes = {
    isPrimary: PropTypes.bool,
    onClick: PropTypes.func.isRequired
};