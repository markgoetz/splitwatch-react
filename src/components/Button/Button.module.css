.button {
  font-family: var(--font-sans);
  font-weight: bold;
  display: block;
  border-radius: 5px;
  text-align: center;
  color: var(--darkest-grey);
  border: none;
  cursor: pointer;
}

.primary {
  background-color: var(--primary-color);
  font-size: 35px;
  box-shadow: 0 4px var(--primary-color-shade);
  padding: 80px;
}

.secondary {
  background-color: var(--medium-grey);
  font-size: 20px;
  box-shadow: 0 4px var(--medium-grey-shade);
  padding: 15px;
}