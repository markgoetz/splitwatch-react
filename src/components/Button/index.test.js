import React from 'react';
import Button from './Button';
import { shallow } from 'preact-render-spy';

describe('Button Component', () => {
  test('renders correctly in primary', () => {
    const context = shallow(<Button isPrimary />);
    expect(context).toMatchSnapshot();
  });

  test('renders correctly when not primary', () => {
    const context = shallow(<Button />);
    expect(context).toMatchSnapshot();
  });
});