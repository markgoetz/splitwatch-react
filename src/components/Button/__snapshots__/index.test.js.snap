// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly in primary 1`] = `
preact-render-spy (1 nodes)
-------
<button class=" "></button>

`;

exports[`Button Component renders correctly when not primary 1`] = `
preact-render-spy (1 nodes)
-------
<button class=" "></button>

`;
